docs/README.md

```diff
@@ -603,33 +620,36 @@ Creates a new ProgramCall object,
 `<object>` a ProgramCall object.
 
 
-## ProgramCall.addParam(value, [options])
+## ProgramCall.addParam(parameter)
 
 **Description:**
 
 Add a new parameter element to the ProgramCall.
 
-**Syntax 1:**
-
-addParam(data, type[,options])
-
-**Syntax 2:**
+**Syntax:**
 
-addParam(data, [,options])
+addParam(parameter)
 
 **Parameters:**
 
-- **value** `<string> | <array>` The parameter element value.
-
-  If the parameter is a data structure, then use syntax 2 where data is an 2D `<array>`.
-
-  Each element of the array can have: `[data, type, options]`.
-
-  Otherwise use syntax 1 where data is the value of the parameter as a `<string>`.
-  
-- **type** `<string>` The parameter data type.
-
-- **[options]** `<object>` additional configuration for the parameter.<br>Refer to the [param, ds, and data tag](http://youngiprofessionals.com/wiki/index.php/XMLSERVICE/XMLSERVICEQuick) for available options.
+- **parameter** `<object>` the parameter.
+
+| Key | Type | Description |
+| --  |  --  |  --   |
+| name | string | This is the name of the parameter |
+| io  | string | parm option to define parameter input output type<br>Valid values are `in`, `out`, or `both` |
+| by  | string | parm option to specify to pass the parameter by reference or value<br>Valid values are `ref` or `val`|
+| type | string | [xmlservice data type](http://yips.idevcloud.com/wiki/index.php/XMLService/DataTypes) or `ds` for a data structure |
+| value | string | This is the value of the data node |
+| fields | array | This is an array of data objects `{ name: type: value: , etc }` used to define ds elements |
+| dim | string | ds option for array dimension value |
+| dou | string | ds option for label match array dou terminate parm label |
+| len | string | ds option for label match calculate length of ds parm label |
+| varying | string | data option for character varying data<br>Valid vlues are `on`, `off`, `2`, `4` |
+| enddo | string | data option for label match array dou terminate parm|
+| setlen | string | data option for label match calculate length of ds parm label |
+| hex | string | data option for hex input character |
+| trim | string | data option to trim character<br> Valid values are `on` or `off`|
```

Comment:
Wouldn't it make more sense to put this in the jsdocs and generate it?

Reply:
I agree will just remove this from the PR.